packages/framer-motion/src/projection/styles/__tests__/transform.test.ts

```diff
@@ -35,4 +37,46 @@ describe("buildProjectionTransform", () => {
             "translate3d(50px, 600px, 0) scale(0.5, 2) rotate(45deg) scale(4, 2)"
         )
     })
+
+    it("Doesn't apply unneccessary translation", () => {
+        const delta = {
+            x: {
+                translate: 0,
+                scale: 2,
+                origin: 0.5,
+                originPoint: 100,
+            },
+            y: {
+                translate: 0,
+                scale: 4,
+                origin: 0.5,
+                originPoint: 100,
+            },
+        }
+
+        expect(
+            buildProjectionTransform(delta, { x: 2, y: 4 }, { rotate: 10 })
+        ).toEqual("scale(0.5, 0.25) rotate(10deg) scale(4, 16)")
+    })
+
+    it("Doesn't apply unneccessary tree scale", () => {
+        const delta = {
+            x: {
+                translate: 100,
+                scale: 2,
+                origin: 0.5,
+                originPoint: 100,
+            },
+            y: {
+                translate: 100,
+                scale: 4,
+                origin: 0.5,
+                originPoint: 100,
+            },
+        }
+
+        expect(
+            buildProjectionTransform(delta, { x: 1, y: 1 }, { rotate: 10 })
+        ).toEqual("translate3d(100px, 100px, 0) rotate(10deg) scale(2, 4)")
+    })
```

Comment:
This is not testing the unnecessary scale, right?

Reply:
Ah - the unnecessary *tree* scale - there are two, one for the element itself (you can see on the `delta`) and one to correct the tree scale (this is the `x: 1 y: 1`).

One has to be applied before and one after `rotate` to ensure we are correcting on the axis aligned with the parent.